### DISCUSSION TITLE
Best practices for saving a tissue classifier object?

### DISCUSSION THREAD

**Discussion by dPys** [dipy/dipy#2228] [Tractography / Tractometry] [answered]
<dipy.tracking.stopping_criterion.BinaryStoppingCriterion at 0x7fce0c4a1730> doesn't seem to like to agree with any form of io. nifti/array-saving methods don't seem to work, pickling it leads to `no default __reduce__ due to non-trivial __cinit__`, and hdf5 can't seem to serialize it either. Has anyone come up with a clean way to do this yet?

Cheers,
@dPys

**Comment by skoudoro** [comment]
Hi @dPys,

I have not try to save the Tissue Classifier so no idea... Maybe @gabknight or @frheault will have a better idea...

**Comment by gabknight** [comment]
Hi @dPys, @skoudoro 

I never saved StoppingCriterion/TissueClassifier either. Is there a specific reason you want to save it? My intuition would be to rather save the input volumes in Nifti format (e.g. mask, metric_map, include_map, exclude_map). 

I think to save them, we would need to implement the __reduce__ function for those cython classes.

**Comment by dPys** [comment, marked as answer]
Thanks @skoudoro and @gabknight , it might have been convenient if the reduce had already existed, but I guess probably not worth the effort to implement it. Will just save everything as a 4d image.  

Cheers,
@dPys
